src/main/java/org/sonarlint/intellij/finding/persistence/LiveFindingCache.java

```diff
@@ -155,12 +152,12 @@ private void flushAll() {
   /**
    * Clear cache and underlying persistent store
    */
-  public synchronized void clear() {
+  public void clear() {
     persistence.clear();
     cache.clear();
   }
 
-  public synchronized boolean contains(VirtualFile virtualFile) {
+  public boolean contains(VirtualFile virtualFile) {
     return getLive(virtualFile) != null;
```

Comment:
Optimization, in the `getLive()` method we are creating a new ArrayList, while we don't actually use it here.

Reply:
Fixed